### ISSUE TITLE
feat: Implement the TreeWalker API

### ISSUE THREAD

**Issue by Hunam6** [b-fuze/deno-dom#35]
Ref: https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker

**Comment by b-fuze** [comment]
Yeah, it's on the list of things to implement when I get some time.

**Comment by Danny-Engelman** [comment]
@b-fuze , I presume you didn't do https://developer.mozilla.org/en-US/docs/Web/API/NodeIterator ?

TreeWalker is the faster way to walk the DOM, NodeIterator only slightly slower

Traversing is SLOW

https://www.measurethat.net/Benchmarks/Show/5581/2/traverse-function-vs-nodeiterator-vs-treewalker#latest_results_block

![](https://i.imgur.com/wereF6b.png)

**Comment by b-fuze** [comment]
@Danny-Engelman no, I didn't do either of them. But TreeWalker is on the plan to to do list.

**Comment by 0xAozora** [comment]
https://github.com/b-fuze/deno-dom/pull/37

**Comment by 0xAozora** [comment]
#37

**Comment by chances** [comment]
This issue is blocking my integration of [`lit-html`](https://lit.dev) in a project.
